diff -Nur GraphicsMagick-1.2.1.orig/configure.ac GraphicsMagick-1.2.1/configure.ac
--- GraphicsMagick-1.2.1.orig/configure.ac	2008-05-05 00:14:52.000000000 +0100
+++ GraphicsMagick-1.2.1/configure.ac	2008-05-14 21:58:25.000000000 +0100
@@ -2744,7 +2744,7 @@
 # Determine path to pick up GraphicsMagick library from for use with building PerlMagick
 MAGICKLIBDIR="${LIB_DIR}"
 MAGICKLIB="-L${MAGICKLIBDIR} -lGraphicsMagick"
-if test $with_perl_static = 'yes'
+if test $with_perl_static = 'yes' -o $with_perl_dynamic = 'yes'
 then
   # Find out where libtool hides its uninstalled libraries (as libtool_objdir)
   libtool_objdir=$objdir
diff -Nur GraphicsMagick-1.2.1.orig/PerlMagick/Makefile.am GraphicsMagick-1.2.1/PerlMagick/Makefile.am
--- GraphicsMagick-1.2.1.orig/PerlMagick/Makefile.am	2007-10-07 01:52:07.000000000 +0100
+++ GraphicsMagick-1.2.1/PerlMagick/Makefile.am	2008-05-14 22:03:00.000000000 +0100
@@ -39,11 +39,12 @@
 	cd $(PERLMAGICK) && @PERL@ Makefile.PL $(PERL_MAKE_OPTIONS)
 
 install-exec-perl: $(PERLMAKEFILE)
-	( cd $(PERLMAGICK) && $(MAKE) CC='@CC@' && \
+	( cd $(PERLMAGICK) && \
 	$(MAKE) CC='@CC@' install && \
 	$(MAKE) clean && rm -f  Makefile.old )
 
-all-perl:
+all-perl: $(PERLMAKEFILE) magick/libGraphicsMagick.la
+	cd $(PERLMAGICK) && $(MAKE) CC='@CC@'
 
 uninstall-exec-perl: $(PERLMAKEFILE)
 	echo "Uninstall not supported for PerlMagick"
