--- GraphicsMagick-1.1.7/configure.ac.orig	2005-09-28 17:05:32.000000000 +0000
+++ GraphicsMagick-1.1.7/configure.ac	2005-09-28 17:06:18.000000000 +0000
@@ -293,7 +293,7 @@
 
 # Configure libltdl
 #AC_CONFIG_SUBDIRS(ltdl)
-AC_LIBLTDL_CONVENIENCE(ltdl)
+AC_LIBLTDL_INSTALLABLE(ltdl)
 AC_LIB_LTDL
 # Substitute INCLTDL and LIBLTDL in the Makefiles
 #AC_SUBST(INCLTDL)
--- GraphicsMagick-1.1.7/utilities/Makefile.am.orig	2005-04-28 21:54:38.000000000 +0200
+++ GraphicsMagick-1.1.7/utilities/Makefile.am	2006-01-28 22:13:21.676806750 +0100
@@ -26,7 +26,7 @@
 
 # Libtool libraries, dependencies, and linker flags used by most executables
 if WITH_LTDL
-COMMONLDADD	= $(MAGICK_LIB) $(top_builddir)/ltdl/libltdlc.la
+COMMONLDADD	= $(MAGICK_LIB) $(LIBLTDL)
 COMMONLDFLAGS	= @LDFLAGS@ @DLLDFLAGS@
 else
 COMMONLDADD	= $(MAGICK_LIB)
