--- GraphicsMagick-1.3.36/configure.ac.orig	2021-12-05 14:39:32.460911190 +0100
+++ GraphicsMagick-1.3.36/configure.ac	2021-12-05 14:42:24.953215236 +0100
@@ -3691,7 +3691,22 @@ MAGICK_EXTRA_DEP_LIBS=`echo $MAGICK_EXTR
 
 MAGICK_API_CFLAGS=$CFLAGS
 MAGICK_API_CPPFLAGS=`echo $MAGICK_API_CPPFLAGS | sed -e 's/  */ /g'`
-MAGICK_API_LDFLAGS="-L$LIB_DIR $LDFLAGS"
+MAGICK_API_LDFLAGS=
+new_LDFLAGS=
+for flag in $LDFLAGS ; do
+  case "$flag" in
+    -L/usr/lib|-L/usr/lib64|-L/usr/libx32|-R/usr/lib|-R/usr/lib64|-R/usr/libx32)
+      ;;
+    -s|-Wl,-s|--strip-all|-Wl,--strip-all|-Wl,--as-needed|-Wl,--no-copy-dt-needed-entries)
+      new_LDFLAGS="$new_LDFLAGS $flag"
+      ;;
+    *)
+      MAGICK_API_LDFLAGS="$MAGICK_API_LDFLAGS $flag"
+      new_LDFLAGS="$new_LDFLAGS $flag"
+      ;;
+  esac
+done
+LDFLAGS="$new_LDFLAGS"
 MAGICK_API_DEP_LIBS="$MAGICK_DEP_LIBS"
 MAGICK_API_LIBS="-lGraphicsMagick $MAGICK_API_DEP_LIBS $MAGICK_EXTRA_DEP_LIBS"
 
